doc/HttpTransportAdapter.md

```diff
@@ -0,0 +1,118 @@
+
+## Default HTTP Transport Adapter
+The Azure SDK for C++ provides two HTTP transport adapters as part of the azure-core package:
+- The libcurl transport adapter
+- WinHttp transport adapter
+
+Azure SDK for C++ users can create and use their own transport adapter as well [see below](#Building-a-custom-HTTP-Transport-Adapter).
+
+For the simplest case, when there was not any specific configuration set for the Azure SDK for C++, the default transport adapter will be selected based on the OS. The WinHTTP transport adapter will be used for Windows-based systems (making libcurl an optional dependency). The libcurl transport adapter will be selected for any non-windows-based system.
+
+## Building HTTP transport adapter
+The Azure SDK for C++ uses CMake options to define what HTTP transport adapters to build.
+
+You can see [CMake options](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#cmake-build-options) to learn about all the supported options and what is the specific option required for each HTTP transport adapter.
+
+Multiple HTTP transport adapters can be built as part of the project. For example, if you want to send HTTP request with the libcurl transport adapter for some cases and the WinHTTP transport adapter for another cases.
+Note that combining WinHTTP transport adapter with another transport adapter is only possible when using Windows.
+
+Another example is if you want to create your own HTTP transport adapter for some other C++ HTTP stack. In this case, you can build your own HTTP transport adapter and also the libcurl HTTP transport adapter and run tests to compare the correctness of both. 
+
+## Using the HTTP transport adapter
+The HTTP transport adapter is set up when a service SDK client is init. For example, when creating an Azure Storage SDK client. The HTTP transport adapter is part of the client options argument that can be used to init the client. See the next example.
+```cpp
+  /*
+  * Using the HTTP transport adapter.
+  */
+
+  // When no options are provided, the HTTP transport adapter is set to the default one.
+  auto defaultStorageClient = BlobServiceClient(url, credential);
+
+  // Use client options to set an HTTP transport adapter.
+  BlobClientOptions options;
+  options.TransportPolicyOptions.Transport = std::make_shared<Azure::Core::Http::CurlTransport>();
+  auto storageClient = BlobServiceClient(url, credential, options);
+```
+
+### Re-use the HTTP transport adapter 
+Note that the HTTP transport adapter is a `shared_ptr`. This is because you can re-use the same HTTP transport adapter for multiple clients. There are two ways of doing this. The first one is by using the same HTTP transport adapter to init a second client. The second one is when the client can create a new client directly witch inherits the same configuration. See the next example:
```

Comment:
```suggestion
Note that the HTTP transport adapter is a `shared_ptr`. This is because you can re-use the same HTTP transport adapter for multiple clients. There are two ways of doing this. The first one is by using the same HTTP transport adapter when creating another client. The second one is when you get a specific child client from an already created parent client directly, in which case the child client inherits the same configuration. See the next example:
```